src/apyfloatarray_wrapper.cc

```diff
@@ -522,7 +522,232 @@ void bind_float_array(nb::module_& m)
+            -------
+            ValueError
+                If given an axis of a size other than 1 a ValueError will be thrown.
+            IndexError
+                If a specified axis is outside of the existing number of dimensions for the array.
+
+            )pbdoc"
+        )
+
+        .def(
+            "sum",
+            &APyFloatArray::sum,
+            nb::arg("axis") = nb::none(),
+            R"pbdoc(
+            Returns the cumulative sum of the elements along specified axis/axes.
+
+            Parameters
+            ----------
+            axis : tuple, int, optional
+                The axis/axes to summate across. Will summate the whole array if no int or tuple is specified.
+
+            Returns
+            -------
+            :class:`APyFloatArray`
+
+            Raises
+            -------
+            IndexError
+                If a specified axis is outside of the existing number of dimensions for the array.
+
+            Examples
+            -------
+
+            >>> from apytypes import APyFloatArray
+
+            Array `a`, array to summate across.
+
+            >>> a = APyFloatArray(
+            ...     [1,2,3,4,5,6],
+            ...     exp_bits=5,
+            ...     man_bits=0
+            ... )
+
+            Array `b`, returned array where the specified axes are summated.
+            >>> b = a.sum()
+            In this case `b` will be equal to `c`.
+            >>> c = APyFloatArray(
+            ...     [21],
+            ...     exp_bits=8,
+            ...     man_bits=0
+            ... )
+
+            )pbdoc"
+        )
+
+        .def(
+            "cumsum",
+            &APyFloatArray::cumsum,
+            nb::arg("axis") = nb::none(),
+            R"pbdoc(
+            Returns the cumulative sum of the elements along a given axis.
+
+            Parameters
+            ----------
+            axis : int, optional
+                The axis to summate across. If not given an axis it will return the cumulative sum of the flattened array.
+
+            Returns
+            -------
+            :class:`APyFloatArray`
+
+            Raises
+            -------
+            IndexError
+                If a specified axis is outside of the existing number of dimensions for the array.
+
+            Examples
+            -------
+
+            >>> from apytypes import APyFloatArray
+
+            Array `a`, array to summate across.
+
+            >>> a = APyFloatArray(
+            ...     [[1,2,3],[4,5,6]],
+            ...     exp_bits=5,
+            ...     man_bits=0
+            ... )
+            Array `b`, returned array where the specified axes are summated.
+            >>> b = a.cumsum()
+            In this case `b` will be equal to `c`.
+            >>> c = APyFloatArray(
+            ...     [1,3,6,10,15,21],
+            ...     exp_bits=8,
+            ...     man_bits=0
+            ... )
+            >>> d = a.cumsum(0)
+            In this case `d` will be equal to `e`.
```

Comment:
There will be a lot of quantization when using 0 mantissa bits (i.e. only powers of 2), will they in fact equal? Using 0 mantissa bits is also not very well tested. Best options is probably to just use a few more mantissa bits ($\geq 4$) for all the examples.

Reply:
i meant to use 10 mantissa bits and 10 exponent bits. It is resolved now